synapse/storage/database.py

```diff
@@ -2195,12 +2199,49 @@ def simple_delete_many_txn(
             clauses.append("%s = ?" % (key,))
             values.append(value)
 
-        if clauses:
-            sql = "%s WHERE %s" % (sql, " AND ".join(clauses))
+        sql = "%s WHERE %s" % (sql, " AND ".join(clauses))
         txn.execute(sql, values)
 
         return txn.rowcount
 
+    async def simple_truncate(self, table: str, desc: str) -> None:
```

Comment:
Does the word `simple` actually do anything here? (What would a complex truncate look like?)

Reply:
Hmm, I'm not sure! I guess it saves you needing to write the SQL yourself?

There's no other operational methods in this class without `simple_`, so we could either start a new trend or keep on with the consistency.

As someone who's worked with Synapse quite a bit, I'd *expect* to see `db_pool.simple_truncate`, and often `db_pool.simpl<tab>` for possibilities. But I'm also obviously biased.